calico/datamodel_v1.py

```diff
@@ -109,7 +109,36 @@ def get_profile_id_for_profile_dir(key):
     return final_node if prefix == PROFILE_DIR else None
 
 
-class EndpointId(namedtuple("EndpointId", ["host", "orchestrator",
-                                           "workload", "endpoint"])):
+class EndpointId(object):
+    __slots__ = ["host", "orchestrator", "workload", "endpoint"]
+
+    def __init__(self, host, orchestrator, workload, endpoint):
+        # Host and orchestrator can be sensibly interned, they appear
+        # regularly.  They shouldn't contain any utf-8 characters but we get
+        # back a unicode string so encode it.
+        self.host = intern(host.encode("utf8"))
+        self.orchestrator = intern(orchestrator.encode("utf8"))
+        # No sensible interning to be done for workload/endpoint.
+        self.workload = workload.encode("utf8")
+        self.endpoint = endpoint.encode("utf8")
+
```

Comment:
Why no interning of endpoint ID / workload ID? Seems to me that we might have multiple endpoints in the same workload (say) - and it feels like it can't hurt. Is there a cost to interning that I'm missing?